drivers/include/ili9341.h

```diff
@@ -0,0 +1,174 @@
+ * @ref ILI9341_LE_MODE compile time option can switch this, but only use this
+ * when strictly necessary. This option will slow down the driver as it
+ * certainly can't use DMA anymore, every short has to be converted before
+ * transfer.
+ */
+
+
+#ifndef ILI9341_H
+#define ILI9341_H
+
+#include "periph/spi.h"
+#include "periph/gpio.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifndef ILI9341_GVDD
+/**
+ * @brief ili9341 gvdd level.
+ *
+ * Default GVDD voltage of 4.8V
+ */
+#define ILI9341_GVDD    4800
+
+#endif /* ILI9341_GVDD */
+
+#ifndef ILI9341_VCOMH
+/**
+ * @brief ili9341 VCOMH voltage level.
+ *
+ * Default VCOMH voltage of 4.25V
+ */
+#define ILI9341_VCOMH   4250
+
+#endif /* ILI9341_VCOMH */
+
+#ifndef ILI9341_VCOML
+/**
+ * @brief ili9341 VCOML voltage level.
+ *
+ * Default VCOMH voltage of -2V
+ */
+#define ILI9341_VCOML   -2000
+
+#endif /* ILI9341_VCOML */
+
+#ifndef ILI9341_LE_MODE
+/**
+ * @brief Compile time switch to change the driver to convert little endian
+ *        colors to big endian.
+ */
+#define ILI9341_LE_MODE     (0)
+#endif
+
+/**
+ * @brief   Device initialization parameters
+ */
+typedef struct {
+    spi_t spi;          /**< SPI device that the display is connected to */
+    spi_clk_t spi_clk;  /**< SPI clock speed to use */
+    gpio_t cs_pin;      /**< pin connected to the CHIP SELECT line */
+    gpio_t dcx_pin;     /**< pin connected to the DC line */
+    gpio_t rst_pin;     /**< pin connected to the reset line */
+} ili9341_params_t;
+
+/**
+ * @brief   Device descriptor for a ili9341
+ */
+typedef struct {
+    ili9341_params_t params;          /**< I2C device that sensor is connected to */
+} ili9341_t;
+
+
+/**
+ * @brief   Setup an ili9341 display device
+ *
+ * @param[out]  dev     device descriptor
+ * @param[in]   params  parameters for device initialization
+ */
+int ili9341_init(ili9341_t *dev, const ili9341_params_t *prms);
+
+/**
+ * @brief   Fill a rectangular area with a single color
+ *
+ * @param[in]   dev     device descriptor
+ * @param[in]   x1      x coordinate of the first corner
+ * @param[in]   x2      x coordinate of the opposite corner
+ * @param[in]   y1      y coordinate of the first corner
+ * @param[in]   y2      y coordinate of the opposite corner
+ * @param[in]   color   single color to fill the area with
+ */
+void ili9341_fill(ili9341_t *dev, uint16_t x1, uint16_t x2, uint16_t y1, uint16_t y2, uint16_t color);
+
+/**
+ * @brief   Fill a rectangular area with an array of colors
+ *
+ * @param[in]   dev     device descriptor
+ * @param[in]   x1      x coordinate of the first corner
+ * @param[in]   x2      x coordinate of the opposite corner
+ * @param[in]   y1      y coordinate of the first corner
+ * @param[in]   y2      y coordinate of the opposite corner
+ * @param[in]   color   array of colors to fill the area with
+ */
+void ili9341_map(ili9341_t *dev, uint16_t x1, uint16_t x2, uint16_t y1,
```

Comment:
I don't think the function name is clear. Maybe `set_area` is clearer?

Reply:
Additional observation: `x2/y2` seem to be the the x/y-coordinate of the pixel in the opposite corner, not the opposite corner itself.

Reply:
> I don't think the function name is clear. Maybe `fill_pixels` is clearer?

Ack

> Additional observation: `x2/y2` seem to be the the x/y-coordinate of the pixel in the opposite corner, not the opposite corner itself (same for all other x2/y2's).

So a documentation fix?

Reply:
If explained clearly a documentation fix is fine to me, but a code change would enable users to do things like: `fill_pixels(x, y, x + picture_width, y + picture_height, picture)`.

Reply:
Renamed to pixmap and added some docs